### PR TITLE
Re-add dynamic delivery_method for ApplicationMailer

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Bug Fix

## Description
The reason my previous attempt in #13943 did not work is most likely due setting `smtp_settings` improperly in the callback. This PR should fix with by altering `delivery_method.settings` instead.

## Related Tickets & Documents
https://github.com/forem/forem/pull/13943

## QA Instructions, Screenshots, Recordings
n/a

## Added/updated tests?
- [x] Yes

## [Forem core team only] How will this change be communicated?
- [x] This change does not need to be communicated yet, will update the doc post POC

## [optional] Are there any post deployment tasks we need to perform?
QA on Canary first.

### CODE REVIEW

File: app/workers/follows/send_email_notification_worker.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def perform(follow_id, mailer = NotifyMailer.name)
 
       return if EmailMessage.where(user_id: follow.followable_id)
         .where("sent_at > ?", rand(15..35).hours.ago)
-        .exists?(["subject LIKE ?", "%#{NotifyMailer.subjects[:new_follower_email]}"])
+        .exists?(["subject LIKE ?", "%#{NotifyMailer.new.subjects[:new_follower_email]}"])
```
Comment (maestromac) [inline app/workers/follows/send_email_notification_worker.rb:12]: I unintentionally found out that this is a syntax error and this code may have been broken the whole time.

File: app/mailers/application_mailer.rb [hunk 2]
```diff
@@ -48,6 +46,6 @@ def set_perform_deliveries
   protected
 
   def set_delivery_options
-    self.smtp_settings = Settings::SMTP.settings
```
Comment (maestromac) [inline app/mailers/application_mailer.rb:51]: Turns out this is the incorrect way to update SMTP config. In spec, checking `ActionMailer::Base.deliveries.last.deliver_methods` after any email is sent out would return `{}`.

**Comment by maestromac** [comment]
Keeping it draft for now to test it on Canary first.

File: app/workers/follows/send_email_notification_worker.rb (hunk 1 above)
Comment (citizen428) [reply app/workers/follows/send_email_notification_worker.rb:12]: @maestromac A syntax error? Not a `NoMethodError`? 😕

Comment (maestromac) [reply app/workers/follows/send_email_notification_worker.rb:12]: @citizen428 ah yup I meant `NoMethodError`

File: app/mailers/concerns/deliverable.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module Deliverable
```
Comment (maestromac) [inline app/mailers/concerns/deliverable.rb:1]: This is more like a placeholder. Happy to take any naming suggestion here

**Comment by andygeorge** [comment]
Deployed this to canary1

**Review by rhymes — APPROVED** [review]
Let's do this!

File: app/mailers/concerns/deliverable.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    before_action :set_perform_deliveries
+    after_action  :set_delivery_options
+  end
+
+  def set_perform_deliveries
+    self.perform_deliveries = ForemInstance.smtp_enabled?
+  end
+
+  def set_delivery_options
+    mail.delivery_method.settings.merge!(Settings::SMTP.settings)
```
Comment (jgaskins) [inline app/mailers/concerns/deliverable.rb:14]: How does this incorporate `SENDGRID_API_KEY`?

Comment (maestromac) [reply app/mailers/concerns/deliverable.rb:14]: `Settings::SMTP.settings` returns a hardcoded config if `SENDGRID_API_KEY` exists.

See https://github.com/forem/forem/blob/main/app/models/settings/smtp.rb#L20

File: spec/mailers/application_mailer_spec.rb [hunk 5]
```diff
@@ -4,7 +4,7 @@
   let(:user) { create(:user) }
   let(:email) { VerificationMailer.with(user_id: user.id).account_ownership_verification_email }
 
-  xdescribe "#set_perform_deliveries" do
+  describe "#set_perform_deliveries" do
```
Comment (djuber) [inline spec/mailers/application_mailer_spec.rb:7]: nitpick, non-blocking: I think this is a Deliverable test, not specifically an ApplicationMailer test.

**Review by djuber — APPROVED** [review]

**Pull request merged**